numpy/_core/numerictypes.py

```diff
@@ -223,10 +223,7 @@ def issctype(rep):
         return False
     try:
         res = obj2sctype(rep)
-        if res and res != object_:
-            return True
-        else:
-            return False
+        return res and res != object_
```

Comment:
This isn't the same, the return type changes, it was bool.

Reply:
Agreed, let's just skip rule 103 at least until it is fixed (although, once you add a bool on top, I lean towards most of this code is being better without the change).

114 (merging ifs with identical branches) isn't quite as nice to me as I would have expected, but if @charris likes it, happy to go with it.

(401 and 910, the dict changes are great IMO)

Reply:
This was a manual change. I keep forgetting that unlike `&&` in C, the result of `and` in Python is not necessarily a bool.

Reply:
Overall, I have mixed feelings about the `SIM` rules as well. Many of these rules may improve or worsen readability, depending on the original code. It's not a good idea to apply them automatically. Let me try to remove the more debatable rules.